mozilla_django_oidc_db/setupconfig/boostrap.py

```diff
@@ -0,0 +1,88 @@
+from django.conf import settings
+from django.contrib.auth.models import Group
+
+from django_setup_configuration.configuration import BaseConfigurationStep
+from django_setup_configuration.exceptions import ConfigurationRunFailed
+
+from ..forms import OpenIDConnectConfigForm
+from ..models import OpenIDConnectConfig
+
+
+class AdminOIDCConfigurationStep(BaseConfigurationStep):
+    """
+    Configure admin login via OpenID Connect
+    """
+
+    verbose_name = "Configuration for admin login via OpenID Connect"
+    required_settings = [
+        "ADMIN_OIDC_OIDC_RP_CLIENT_ID",
+        "ADMIN_OIDC_OIDC_RP_CLIENT_SECRET",
+        "ADMIN_OIDC_OIDC_OP_AUTHORIZATION_ENDPOINT",
+        "ADMIN_OIDC_OIDC_OP_TOKEN_ENDPOINT",
+        "ADMIN_OIDC_OIDC_OP_USER_ENDPOINT",
+    ]
+    all_settings = required_settings + [
+        "ADMIN_OIDC_OIDC_RP_SCOPES_LIST",
+        "ADMIN_OIDC_OIDC_RP_SIGN_ALGO",
+        "ADMIN_OIDC_OIDC_RP_IDP_SIGN_KEY",
+        "ADMIN_OIDC_OIDC_OP_DISCOVERY_ENDPOINT",
+        "ADMIN_OIDC_OIDC_OP_JWKS_ENDPOINT",
+        "ADMIN_OIDC_USERNAME_CLAIM",
+        "ADMIN_OIDC_GROUPS_CLAIM",
+        "ADMIN_OIDC_CLAIM_MAPPING",
+        "ADMIN_OIDC_SYNC_GROUPS",
+        "ADMIN_OIDC_SYNC_GROUPS_GLOB_PATTERN",
+        "ADMIN_OIDC_DEFAULT_GROUPS",
+        "ADMIN_OIDC_MAKE_USERS_STAFF",
+        "ADMIN_OIDC_SUPERUSER_GROUP_NAMES",
+        "ADMIN_OIDC_OIDC_USE_NONCE",
+        "ADMIN_OIDC_OIDC_NONCE_SIZE",
+        "ADMIN_OIDC_OIDC_STATE_SIZE",
+        "ADMIN_OIDC_USERINFO_CLAIMS_SOURCE",
+    ]
+    enable_setting = "ADMIN_OIDC_CONFIG_ENABLE"
+
+    def is_configured(self) -> bool:
+        return OpenIDConnectConfig.get_solo().enabled
+
+    def configure(self):
+        config = OpenIDConnectConfig.get_solo()
+
+        # Use the model defaults
+        form_data = {
+            field.name: getattr(config, field.name)
+            for field in OpenIDConnectConfig._meta.fields
+        }
+
+        # `email` is in the claim_mapping by default, but email is used as the username field
+        # by OIP, and you cannot map the username field when using OIDC
+        if "email" in form_data["claim_mapping"]:
+            del form_data["claim_mapping"]["email"]
```

Comment:
this is not an assumption you can make at the library level - it's up to the project to correctly specify/provide the claim mapping and raise `ImproperlyConfigured` if a known invalid configuration is detected

```suggestion
```